pkg/tc/options.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tc
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/codenotary/immudb/pkg/client"
+)
+
+// Options trust checker options
+type Options struct {
+	Address       string
+	Port          int
+	ImmudbAddress string
+	ImmudbPort    int
+	Detached      bool
+	MTLs          bool
+	MTLsOptions   client.MTLsOptions
+	Config        string
+	Pidfile       string
+	Logfile       string
+}
+
+// DefaultOptions trust checker default options
+func DefaultOptions() Options {
+	return Options{
+		Address:       "127.0.0.1",
+		Port:          3325,
+		ImmudbAddress: "127.0.0.1",
+		ImmudbPort:    3322,
+		Detached:      false,
+		MTLs:          false,
+		Config:        "configs/immutc.toml",
+		Pidfile:       "",
+		Logfile:       "",
+	}
+}
+
+// WithAddress sets address
+func (o Options) WithAddress(address string) Options {
+	o.Address = address
+	return o
+}
+
+// WithPort sets port
+func (o Options) WithPort(port int) Options {
+	o.Port = port
+	return o
+}
+
+// WithImmudbAddress sets immudbAddress
+func (o Options) WithImmudbAddress(immudbAddress string) Options {
+	o.ImmudbAddress = immudbAddress
+	return o
+}
+
+// WithImmudbPort sets immudbPort
+func (o Options) WithImmudbPort(immudbPort int) Options {
+	o.ImmudbPort = immudbPort
+	return o
+}
+
+// WithMTLs sets MTLs
+func (o Options) WithMTLs(MTLs bool) Options {
+	o.MTLs = MTLs
+	return o
+}
+
+// WithDetached sets immutc to be run in background
+func (o Options) WithDetached(detached bool) Options {
+	o.Detached = detached
+	return o
+}
+
+// WithMTLsOptions sets MTLsOptions
+func (o Options) WithMTLsOptions(MTLsOptions client.MTLsOptions) Options {
+	o.MTLsOptions = MTLsOptions
+	return o
+}
+
+// WithConfig sets config
+func (o Options) WithConfig(config string) Options {
+	o.Config = config
+	return o
+}
+
+// WithPidfile sets pidfile
+func (o Options) WithPidfile(pidfile string) Options {
+	o.Pidfile = pidfile
+	return o
+}
+
+// WithLogfile sets logfile
+func (o Options) WithLogfile(logfile string) Options {
+	o.Logfile = logfile
+	return o
+}
+
+func (o Options) Bind() string {
+	return o.Address + ":" + strconv.Itoa(o.Port)
```

Comment:
Can be replaced with fmt.Sprintf instead of converting string and manually parsing the address.